### ISSUE TITLE
Code is incompatible with Boost 1.76

### ISSUE THREAD

**Issue by rtobar** [ska-sa/spead2#139]
When compiling against boost 1.76 one gets the following error:

```
In file included from send_tcp.cpp:24:
../include/spead2/common_socket.h: In instantiation of 'bool spead2::socket_uses_io_service(SocketType&, boost::asio::io_service&) [with SocketType = boost::asio::basic_stream_socket<boost::asio::ip::tcp>; boost::asio::io_service = boost::asio::io_context]':
send_tcp.cpp:178:63:   required from here
../include/spead2/common_socket.h:83:34: error: no match for 'operator==' (operand types are 'boost::asio::basic_socket<boost::asio::ip::tcp, boost::asio::any_io_executor>::executor_type' {aka 'boost::asio::any_io_executor'} and 'boost::asio::io_context::executor_type' {aka 'boost::asio::io_context::basic_executor_type<std::allocator<void>, 0>'})
   83 |     return socket.get_executor() == io_service.get_executor();
      |            ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/boost/asio/detail/io_object_impl.hpp:23,
                 from /usr/local/include/boost/asio/basic_socket.hpp:22,
                 from /usr/local/include/boost/asio/basic_datagram_socket.hpp:20,
                 from /usr/local/include/boost/asio.hpp:24,
                 from ../include/spead2/common_socket.h:29,
                 from send_tcp.cpp:24:
```

Here is the full log: [error.log](https://github.com/ska-sa/spead2/files/6653774/error.log). This happened on a docker image with Ubuntu 21.10, but it was originally spotted by @steve-ord on a Mac.

**Comment by bmerry** [comment]
Thanks for the heads-up. I wish Boost.Asio wouldn't keep changing the API. It will probably be a little while before I have time to dig into the guts of Boost.Asio to figure out how to compare these things, but this function is only used for error checking so if you just want a quick hack to get going you can replace it with `return true;`

**Comment by bmerry** [comment]
It is a Boost.Asio bug: https://github.com/chriskohlhoff/asio/issues/853. I've made a branch called boost-1.76-workaround; let me know if that fixes it for you.

**Comment by rtobar** [comment]
Thanks @bmerry for the very quick investigation and fix. I tried building the library on its own, then as part of the python package, and it all works now. I also noticed you put a fix for using `python3` explicitly everywhere, which tackled a problem I also experienced but didn't report (I worked around it in Ubuntu by installing the `python-is-python3` package).

**Comment by bmerry** [comment]
Fixed in spead2 3.2.2.
